gnome/gobject-introspection/files/patch-fix-rpath-gir-typelib.diff

```diff
@@ -29,7 +29,7 @@
 +            if libdir == '':
 +                # no way to resolve @rpath and the like, since we can't
 +                # make a reasonable assumpion of the install libdir
-+                print("warning: unknown install library directory! GObject Introspection GIR and TYPELIB files might not work!")
++                libdir = '/opt/local/lib/'
```

Comment:
I have no idea if this will work or not for all use cases of GO-I. I would strongly suggest leaving the warning but changing to be more relevant to this setting. The setting must be done in 2 steps: (1) this patch, but change the directory to something like `@MP_LIB@`; (2) in the Portfile add a `post-patch` to `reinplace` the `@MP_LIB` to `${prefix}/lib`. While I'd think the vast vast majority of MP uses do use the default of `/opt/local` for the install prefix, not all do. We want to make sure the correct directory is used, and the only way to do that is to replace it via this 2-step patch method. Once that's done I can test this PR out, but I'm not holding my breath that it is a global fix.

Reply:
Thank you for a detailed answer. I have an example of two libraries fixed by this PR - `gdk-pixbuf2` and `atk`.
I'll try to apply your suggestions to them and see how it goes.

Reply:
@michaelld I pushed the changes, but it doesn't work for `atk` anymore. The previous version also stopped working for `atk`, I don't know the reason, but unfortunately I don't have time to invest in this right now.

Reply:
Thanks for getting this to the split patch+reinplace, @idryzhov . I'm going to merge as is because it does fix the `@rpath@` giscanner detection setting, even if not perfect it's better than what we have at the moment. I'll look into the `atk` port's issue.